### ISSUE TITLE
Intent-to-Deprecate: amp-script "development" mode.

### ISSUE THREAD

**Issue by samouri** [ampproject/amphtml#25476]
## Summary

The `amp-script` extension provides a development mode which allows users to disable the file size limitation. This is desirable because assets are usually unminified during development which means much larger file sizes.

Currently, developers can toggle this mode by adding a `development` attribute to the amp-script element.  We'd like to change this so that developers indicate development mode by adding a `data-ampdevmode` to both the root html element and the relevant `amp-script` tags (technically one may place the data-ampdevmode attribute anywhere above the relevant `amp-script`, matching how the validator behaves).

## Motivation

See https://github.com/ampproject/amphtml/issues/24987 for the full discussion.
tldr: It is confusing that AMP has multiple different development toggles and we'd like to converge onto one.

## Impact on existing users
Very minor changes in the development flow of a subset of amp site developers.

## Alternative implementation suggestion for developers using AMP

Add the `data-ampdevmode` tag to both the root html element and the amp-script element.

/cc @ampproject/wg-approvers, @jridgewell, @choumx

**Comment by cramforce** [comment]
Could this use the `#development=1` dev mode that triggers the validator?

**Comment by samouri** [comment]
See @choumx's response to this in https://github.com/ampproject/amphtml/issues/24444#issuecomment-530144835:

> #development=1 means real users could get this behavior on production AMP pages e.g. bad links/embeds, URL sharing.

**Comment by cramforce** [comment]
SG

**Comment by samouri** [comment]
The documentation has been updated and we've been emitting warnings since early Nov (https://github.com/ampproject/amphtml/pull/25454). I think its fair to fully remove support for the older method of opting in to amp-script dev mode now.

I'll open a PR
